python/Ganga/Core/GangaRepository/SubJobXMLList.py

```diff
@@ -199,13 +204,32 @@ def __iter__(self):
         return SJXLIterator(self)
 
     def __get_dataFile(self, index, force_backup=False):
+
+        backup_decision = self._load_backup is True or force_backup is True
+
+        index_str = str(index)+"_"+str(backup_decision)
+
+        if index_str in self._cached_filenames:
+            return self._cached_filenames[index_str]
+
         import os.path
         subjob_data = os.path.join(self._jobDirectory, str(index), self._dataFileName)
-        if self._load_backup is True or force_backup is True:
+        if backup_decision is True:
             subjob_data = subjob_data + '~'
+
+        self._cached_filenames[index_str] = subjob_data
         return subjob_data
 
     def __len__(self):
+
+        import os
+        this_time = os.stat(self._jobDirectory).st_ctime
+
+        if len(self._stored_len) == 2:
+            last_time = self._stored_len[0]
+            if this_time == last_time:
+                return self._stored_len[1]
+
```

Comment:
What is `_stored_len` and what is it used for. I feel adding code like this should really be accompanied by some comments otehrwise in 5 years, no-one will have a clue (e.g. our conversation in the meeting about the prepared state).